service/tests/server/mod.rs

```diff
@@ -94,15 +103,24 @@ impl Drop for TestServer {
 }
 
 impl TestServer {
-    pub async fn new() -> Self {
+    pub async fn new(tls: Option<TlsConfig>) -> Self {
         let mut test_node = TestPostService::new();
         let reg = test_node.register_for_connections();
 
         let listener = TcpListener::bind("[::1]:0").await.unwrap();
         let addr = listener.local_addr().unwrap();
 
-        let handle = tokio::spawn(
+        let mut server = if let Some(tls) = &tls {
+            let tls = ServerTlsConfig::new()
+                .identity(tls.server.clone())
+                .client_ca_root(tls.client_ca_cert.clone());
+            Server::builder().tls_config(tls).unwrap()
+        } else {
             Server::builder()
+        };
```

Comment:
Please move call to `Server::builder()` out of `if` body (as it's common thing in both code paths) like:
```rust
        let mut server = Server::builder();
        if let Some(tls) = &tls {
            let tls = ServerTlsConfig::new()
                .identity(tls.server.clone())
                .client_ca_root(tls.client_ca_cert.clone());
            server = server.tls_config(tls).unwrap()
        };
```